ai_genomics/analysis/openalex_definition.py

```diff
@@ -0,0 +1,630 @@
+        all_ai_mesh,
+        mesh_genomics_terms,
+        concept_name="descriptor_name",
+    )
+
+    logging.info(f"AI with genomics mesh {len(ai_works_mesh_genomics)}")
+
+    overlap = len(
+        set(ai_works_meta_genomics["work_id"]).union(
+            set(ai_works_mesh_genomics["work_id"])
+        )
+    )
+
+    logging.info(f"AI with genomics concept or mesh {overlap}")
+
+    logging.info("Crude genomic abstract search")
+    non_empty_abstracts = {k: v for k, v in all_abstracts.items() if type(v) == str}
+    crude_genomic_search = {
+        k: v for k, v in non_empty_abstracts.items() if "genom" in v
+    }
+
+    all_genomics_ids = (
+        set(ai_works_meta_genomics["work_id"])
+        .union(set(ai_works_mesh_genomics["work_id"]))
+        .union(set(crude_genomic_search.keys()))
+    )
+    ai_genomics_all_approaches = all_works_provisional.loc[
+        all_works_provisional["work_id"].isin(all_genomics_ids)
+    ].reset_index(drop=True)
+
+    logging.info(
+        f"AI with genomics concept through all search strategies {len(ai_genomics_all_approaches)}"
+    )
+
+    logging.info("Crude search for AI papers in the genetics dataset")
+    ai_terms_genetics = config["ai_terms_genetics"]
+
+    non_empty_genetics = {k: v for k, v in abstracts_genetics.items() if type(v) == str}
+    crude_genomic_ai_search = {
+        k: v
+        for k, v in non_empty_genetics.items()
+        if any(t in v for t in ai_terms_genetics)
+    }
+    genetic_ai_abstract_ids = set(crude_genomic_ai_search.keys())
+
+    logging.info(f"Genetics papers with AI terms: {len(genetic_ai_abstract_ids)}")
+
+    logging.info("Look for genetics papers with AI concepts")
+    genetic_ai_definition_ids = pipe(
+        subset_on_concepts(
+            works_meta_genetics_filtered,
+            works_concepts_genetics,
+            {"Artificial intelligence": 0, "Machine learning": 0, "Deep learning": 0},
+            inclusive=True,
+            return_excluded=False,
+        )["work_id"],
+        set,
+    )
+
+    logging.info(f"Genetics papers with AI concepts: {len(genetic_ai_definition_ids)}")
+
+    logging.info("Combine everything")
+    ai_gen_total_ids = all_genomics_ids.union(genetic_ai_abstract_ids).union(
+        genetic_ai_definition_ids
+    )
+
+    logging.info(f"Genetics papers in all approaches: {len(ai_gen_total_ids)}")
+
+    # Provisional dataset
+    ai_genomics_provisional_dataset = (
+        pd.concat(
+            [
+                all_works_provisional.loc[
+                    all_works_provisional["work_id"].isin(ai_gen_total_ids)
+                ],
+                works_meta_genetics.loc[
+                    works_meta_genetics["work_id"].isin(ai_gen_total_ids)
+                ],
+            ]
+        )
+        .drop_duplicates("work_id")
+        .reset_index(drop=True)
+    )
+
+    ai_genomics_provisional_dataset.to_csv(
+        f"{PROJECT_DIR}/outputs/ai_genomics_provisional_dataset.csv"
+    )
```

Comment:
Could `ai_genomics_provisional_dataset` be saved to S3? so that other people can access the file without having to run the script.

Reply:
Let's do this once we have a complete dataset with extra openalex concepts.